File: frontend/src/types/seed.ts
export interface Language {
  id: number;
  code: string;
  name_native: string;
  is_active: boolean;
}

export interface Avatar {
  id: number;
  name: string;
  thumbnail_url: string;
  primary_color: string;
  model_url: string | null;
  is_active: boolean;
}

export interface Voice {
  id: number;
  language_id: number;
  name: string;
  gender: 'male' | 'female';
  tone: string | null;
  sample_url: string | null;
  description: string | null;
  is_active: boolean;
}

export interface LanguageListResponse {
  items: Language[];
}

export interface AvatarListResponse {
  items: Avatar[];
}

export interface VoiceListResponse {
  items: Voice[];
}

export interface CefrLevel {
  code: string;       // A1, A2, B1, B2, C1, C2
  clazz: string;      // 초급, 중급, 고급
  name: string;       // 입문, 초급, 중급, 중상급, 상급, 최상급
  description: string;
  sort_order: number;
}

File: backend/app/schemas/seed.py
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class LanguageResponse(BaseModel):
    id: int
    code: str
    name_native: str
    is_active: bool

    model_config = {"from_attributes": True}


class AvatarResponse(BaseModel):
    id: int
    name: str
    thumbnail_url: str
    primary_color: str
    model_url: Optional[str] = None
    is_active: bool

    model_config = {"from_attributes": True}


class VoiceResponse(BaseModel):
    id: int
    language_id: int
    name: str
    gender: str
    tone: Optional[str] = None
    sample_url: Optional[str] = None
    description: Optional[str] = None
    is_active: bool

    model_config = {"from_attributes": True}


class LanguageListResponse(BaseModel):
    items: List[LanguageResponse]


class AvatarListResponse(BaseModel):
    items: List[AvatarResponse]


class VoiceListResponse(BaseModel):
    items: List[VoiceResponse]


class CefrLevelResponse(BaseModel):
    code: str
    group: str
    name: str
    description: str
    sort_order: int

    model_config = ConfigDict(from_attributes=True)
